components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/PlacesStorage.kt

```diff
@@ -48,4 +52,15 @@ abstract class PlacesStorage(context: Context) : Storage, SyncableStore {
         scope.coroutineContext.cancelChildren()
         places.close()
     }
+
+    /**
+     * Runs [block] described by [operation], ignoring and logging any thrown [UrlParseFailed] exceptions.
+     */
+    protected fun ignoreUrlExceptions(operation: String, block: () -> Unit) {
```

Comment:
nit: `inline` would probably help performance if we hit this case often. No biggie though.